### DISCUSSION TITLE
Pyright doesn't throw errors if I try to use a class method that doesn't exist

### DISCUSSION THREAD

**Discussion by kappa90** [microsoft/pyright#8348] [Q&A] [answered]
Take this code example:

```
sourceset, ready_sourceitems = SourceManagerUtils.remove_boilerplate(
                sourceset,
                ready_sourceitems[:10],
            )
```

If `remove_boilerplate` hasn't been implemented yet, Mypy will throw an error, while Pylance/pyright won't.

I haven't found a way to enforce this behavior, but it would be critical since I'm refactoring a large project.

Any ideas? Thank you.

**Comment by erictraut** [comment]
Could you provide a self-contained, minimal code sample? Your code above references undefined multiple undefined symbols.

My guess, based on the symptoms, is that `SourceManagerUtils` is a class that derives from `Any` or some unknown class, but I can't confirm my theory without additional context.

**Reply by kappa90** [reply]
Sure thank you for answering so quickly. Maybe I'm missing something.

```
class SourcemanagerUtils:
     # various static methods are implemented here
     
    @staticmethod
    def search_sourcesets_by_query_sync(...) -> None
       # some code

    # remove_boilerplate (from the example above) is commented out in the class
    # remove_boilerplate(
    #            sourceset: SourceSet,
    #            ready_sourceitems: list[SourceItem],
    #) -> tuple[SourceSet, list[SourceItem]]
    
# I would expect pyright to tell me here that `remove_boilerplate` does not exist
sourceset, ready_sourceitems = SourceManagerUtils.remove_boilerplate(
                sourceset,
                ready_sourceitems[:10],
            )

**Reply by erictraut** [reply]
The code you've posted above has syntax errors and therefore cannot be cleanly parsed. Also, the name used in the class definition (`SourcemanagerUtils`) is different from the name of the class you reference later (`SourceManagerUtils`). If you fix these issues, pyright will tell you that `remove_boilerplate` doesn't exist.

Code sample in [pyright playground](https://pyright-play.net/?code=MYGwhgzhAEDKD2BXATsApgWTAOzAczWQFUAXASxAgC4Aoae6AAQhLHOAFs0SALeAEzoN%2BaAGbQIaMKh4B9CElRpJJCLIBGAT1kBHRIW0RN2YAAoAlNAC0APmgA5eNjS0Gb6AAdIEGjQUp0FQAaaGQpfkNFdDISNA4YAF44KMwcfEJSCggAOjCOeAA3NA14CkIPcFiLIA)

```python
class SourceManagerUtils:
    @staticmethod
    def search_sourcesets_by_query_sync() -> None:
        pass

SourceManagerUtils.remove_boilerplate()
```

![image](https://github.com/user-attachments/assets/1c6eb13e-d471-4975-9d4e-3658e8fc497d)

**Reply by kappa90** [reply, marked as answer]
Apologies, I wrote that code from memory, not copying the actual code, just to show what my problem is.

I realized that the problem was in the Pylance configuration. The `python.analysis.typeCheckingMode` value had been set to `off` from another extension.

Thank you and sorry for the misunderstanding.
